### PR TITLE
[WIP] quantize conv fusion builder to handle BN folding and f32 weights quantization 

### PR DESCRIPTION
**Pull request by ashokei**

### CODE REVIEW

File: src/ngraph/builder/quantization.hpp [hunk 1]
```diff
@@ -145,5 +145,27 @@ namespace ngraph
                                                     std::shared_ptr<Node> min_freezed_output_conv_2,
                                                     std::shared_ptr<Node> max_freezed_output_conv_2,
                                                     const bool with_relu);
+        std::shared_ptr<Node>
```
Comment (mbrookhart) [inline src/ngraph/builder/quantization.hpp]: Documentation?

File: src/ngraph/builder/quantization.cpp [hunk 2]
```diff
@@ -352,5 +361,172 @@ namespace ngraph
...
+            }
+            // i8.conv -> sum -> relu
+            if (sum_input != nullptr)
+            {
+                sum_min_input = std::make_shared<op::Broadcast>(sum_min_input, q_shape, AxisSet{0});
+                sum_max_input = std::make_shared<op::Broadcast>(sum_max_input, q_shape, AxisSet{0});
+                auto sum_scale = builder::quantization_util::get_sum_scale(
+                    min_freezed_output, max_freezed_output, sum_min_input, sum_max_input);
+
+                // TODO: check for signed or unsigned sum
```
Comment (cconvey) [inline src/ngraph/builder/quantization.cpp]: Does this TODO need to happen in the current PR?

File: src/ngraph/runtime/cpu/builder/quantized_conv.cpp [hunk 3]
```diff
@@ -212,45 +214,58 @@ namespace ngraph
...
-                                    if (old_pops.kind(i) == mkldnn::primitive::kind::sum)
-                                    {
-                                        new_pops.append_sum(dyn_post_op_scales[0]);
-                                    }
+                                    mkldnn::algorithm alg;
+                                    float scale, alpha, beta;
+                                    old_pops.get_params_eltwise(i, scale, alg, alpha, beta);
+                                    new_pops.append_eltwise(scale, alg, alpha, beta);
+                                }
+                                if (old_pops.kind(i) == mkldnn::primitive::kind::sum)
```
Comment (cconvey) [inline src/ngraph/runtime/cpu/builder/quantized_conv.cpp]: For the sake of clarity, would it make sense to either replace this `if` with `else if`, or perhaps use a `switch ... case` statement?

File: src/ngraph/runtime/cpu/builder/quantized_conv.cpp [hunk 4]
```diff
@@ -193,6 +193,8 @@ namespace ngraph
                     auto& arg0_tensor = external_function->get_tensor_data(args[0].get_name());
                     auto& arg1_tensor = external_function->get_tensor_data(args[1].get_name());
                     auto& arg2_tensor = external_function->get_tensor_data(args[2].get_name());
+                    auto& arg3_tensor = external_function->get_tensor_data(args[3].get_name());
+                    size_t arg3_size = args[3].get_size();
                     auto& arg4_tensor = external_function->get_tensor_data(args[4].get_name());
                     auto& arg5_tensor = external_function->get_tensor_data(args[5].get_name());
```
Comment (cconvey) [inline src/ngraph/runtime/cpu/builder/quantized_conv.cpp]: Perhaps out of scope for this PR, but would the code be clearer if these `arg?_tensor` variables were named according to the actual role of each tensor?

Comment (ashokei) [reply src/ngraph/runtime/cpu/builder/quantized_conv.cpp]: @cconvey thanks for review/feedback, this PR is WIP needs cleanup, looks like @jbobba wants this to split up also.

**Comment by ashokei** [comment]
moving this builder/fusion to ngraph-mxnet bridge code.

**Pull request closed without merging**